src/vcpkg-test/files.cpp

```diff
@@ -1029,7 +1029,7 @@ TEST_CASE ("win32_fix_path_case", "[files]")
     CHECK(win32_fix_path_case("\\WiNdOws") == "\\Windows");
     CHECK(win32_fix_path_case("\\WiNdOws") == "\\Windows");
     CHECK(win32_fix_path_case("c:WiNdOws") == "C:Windows");
-    CHECK(win32_fix_path_case("c:WiNdOws/system32") == "C:Windows\\System32");
+    CHECK(win32_fix_path_case("c:WiNdOws/system32") == "C:\\Windows\\system32");
```

Comment:
```suggestion
    CHECK(win32_fix_path_case("c:WiNdOws/system32") == "C:Windows\\system32");
```

Reply:
or
```suggestion
    CHECK(win32_fix_path_case("c:/WiNdOws/system32") == "C:\\Windows\\system32");
```
depending on if the test case is broken or not but this would require `CHECK(win32_fix_path_case("c:WiNdOws") == "C:Windows");` to also be fixed.

Reply:
Ok, I have a fresh Windows 10 Pro, from ISO install, everything in environment variables when referencing just systm32 has it as lower case, ie COMSPEC, Powershell Modules and PATH, but when referencing directories inside system32 it references them in upper case:

echo %PATH%: C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem;C:\Windows\System32\WindowsPowerShell\v1.0\;C:\Windows\System32\OpenSSH\

echo %COMSPEC%
C:\Windows\system32\cmd.exe

echo %PSModulePath%
C:\Program Files\PowerShell\7\Modules;C:\Program Files\WindowsPowerShell\Modules;C:\Windows\system32\WindowsPowerShell\v1.0\Modules;C:\Program Files (x86)\Microsoft SQL Server\150\Tools\PowerShell\Modules\

So I though maybe the ISO was corrupt or something, so I check my old laptop, has not been refreshed since new in 2015, Win10 Enterprise and the same PATH, COMSPEC and  PSModulePath.

So I check my wife's laptop, Win10 Pro, not refreshed just updated, again same results.

So, I said, I changed the system32 to System32 in my path, rebooted and created a new vcpkg install with latest vcpkg-tool and tried `vcpkg install libiconv --debug` and it fail immediately.

I also notice when you open a command prompt (admin) it goes to system32 instead of user directory and my Prompt is setup to show directory, version of windows etc, it show system32 in lower-case also.
User.Name@ANDROMEDA Microsoft Windows [Version 10.0.19044.1566] 03/07/2022 15:13:32
C:\Windows\system32 ->
But if I cd .. then cd system32 it shows system32 in upper case:
User.Name@ANDROMEDA Microsoft Windows [Version 10.0.19044.1566] 03/07/2022 15:20:06
C:\Windows\System32 ->

All is know is the bash command fails when vcpkg passes a Upper-Case System32 to it and succeeds if vcpkg passes a lower-case system32 to it.

I know windows kernel does not care about case, System32, SYSTEM32, system32, SyStEm32, etc - all are the same to it, but a -nix kernel, they are all different directory names.

I will let you all hash it out... 
Is the CI using the 2022-03-03 vcpkg-tool version yet, as last I did a bootstrap-vcpkg.bat on Saturday, it was still pulling the 2022-02-24 version and reviewing some of the CI jobs in pull requests run logs I could see 2022-02-24 was still being used. 
Has anyone tried to use the 2022-03-03 vcpkg-tool version and tried to install port `libiconv` to see if they can duplicate the bash command failing? 
Just let me know how we proceed or if I should delete the PR and close the ticket over on vcpkg.

Reply:
>I know windows kernel does not care about case, System32, SYSTEM32, system32, SyStEm32, etc - all are the same to it, but a -nix kernel, they are all different directory names.

Right, this is exactly why we would have changed the name to match the real one.

I can repro:

```
PowerShell 7.2.1
Copyright (c) Microsoft Corporation.

https://aka.ms/powershell
Type 'help' to get help.

PS C:\Dev> cd vcpkg
PS C:\Dev\vcpkg> git fetch --all --prune
Fetching origin
remote: Enumerating objects: 281, done.
remote: Counting objects: 100% (206/206), done.
remote: Compressing objects: 100% (19/19), done.
remote: Total 281 (delta 188), reused 197 (delta 187), pack-reused 75
Receiving objects: 100% (281/281), 250.48 KiB | 1.15 MiB/s, done.
Resolving deltas: 100% (190/190), completed with 80 local objects.
From github.com:microsoft/vcpkg
   b44c4dd0e..5962cf5a3  master     -> origin/master
 * [new branch]          update-vms -> origin/update-vms
Fetching BillyONeal
PS C:\Dev\vcpkg> git switch -d origin/master
Previous HEAD position was 75ae1ef84 [libssh2] Fix various issues + add features (#23327)
HEAD is now at 5962cf5a3 [mimalloc] Update to 2.0.5 (#23264)
PS C:\Dev\vcpkg> .\bootstrap-vcpkg.bat
Downloading https://github.com/microsoft/vcpkg-tool/releases/download/2022-02-24/vcpkg.exe -> C:\Dev\vcpkg\vcpkg.exe... done.
Validating signature... done.

Telemetry
---------
vcpkg collects usage data in order to help us improve your experience.
The data collected by Microsoft is anonymous.
You can opt-out of telemetry by re-running the bootstrap-vcpkg script with -disableMetrics,
passing --disable-metrics to vcpkg on the command line,
or by setting the VCPKG_DISABLE_METRICS environment variable.

Read more about vcpkg telemetry at docs/about/privacy.md
PS C:\Dev\vcpkg> .\vcpkg.exe install libiconv
Computing installation plan...
The following packages will be built and installed:
    libiconv[core]:x86-windows -> 1.16#11
Detecting compiler hash for triplet x86-windows...
Restored 0 packages from C:\Users\bion\AppData\Local\vcpkg\archives in 472.8 us. Use --debug to see more details.
Starting package 1/1: libiconv:x86-windows
Building package libiconv[core]:x86-windows...
-- Using cached libiconv-1.16.tar.gz.
-- Extracting source C:/Dev/vcpkg-downloads/libiconv-1.16.tar.gz
-- Applying patch 0002-Config-for-MSVC.patch
-- Applying patch 0003-Add-export.patch
-- Applying patch 0004-ModuleFileName.patch
-- Using source at C:/Dev/vcpkg/buildtrees/libiconv/src/1.16-b0c41d3469.clean
-- Getting CMake variables for x86-windows
-- Using cached msys-gzip-1.11-1-x86_64.pkg.tar.zst.
-- Using cached msys-bash-5.1.008-1-x86_64.pkg.tar.zst.
-- Using cached msys-autoconf-2.71-3-any.pkg.tar.zst.
-- Using cached msys-diffutils-3.8-2-x86_64.pkg.tar.zst.
-- Using cached msys-binutils-2.37-5-x86_64.pkg.tar.zst.
-- Using cached msys-libtool-2.4.6-9-x86_64.pkg.tar.xz.
-- Using cached msys-file-5.41-2-x86_64.pkg.tar.zst.
-- Using cached msys-zlib-1.2.11-1-x86_64.pkg.tar.xz.
-- Using cached msys-libbz2-1.0.8-3-x86_64.pkg.tar.zst.
-- Using cached msys-coreutils-8.32-2-x86_64.pkg.tar.zst.
-- Using cached msys-grep-3.0-2-x86_64.pkg.tar.xz.
-- Using cached msys-sed-4.8-2-x86_64.pkg.tar.zst.
-- Using cached msys-libpcre-8.45-1-x86_64.pkg.tar.zst.
-- Using cached msys-m4-1.4.19-2-x86_64.pkg.tar.zst.
-- Using cached msys-automake-wrapper-11-4-any.pkg.tar.zst.
-- Using cached msys-gawk-5.1.0-2-x86_64.pkg.tar.zst.
-- Using cached msys-mpfr-4.1.0-1-x86_64.pkg.tar.zst.
-- Using cached msys-gmp-6.2.1-1-x86_64.pkg.tar.zst.
-- Using cached msys-libreadline-8.1.001-1-x86_64.pkg.tar.zst.
-- Using cached msys-ncurses-6.2-2-x86_64.pkg.tar.zst.
-- Using cached msys-automake1.16-1.16.3-3-any.pkg.tar.zst.
-- Using cached msys-perl-5.32.1-2-x86_64.pkg.tar.zst.
-- Using cached msys-libcrypt-2.1-3-x86_64.pkg.tar.zst.
-- Using cached msys-pkg-config-0.29.2-4-x86_64.pkg.tar.zst.
-- Using cached msys-make-4.3-3-x86_64.pkg.tar.zst.
-- Using cached msys-findutils-4.8.0-1-x86_64.pkg.tar.zst.
-- Using cached msys-libintl-0.21-1-x86_64.pkg.tar.zst.
-- Using cached msys-libiconv-1.16-2-x86_64.pkg.tar.zst.
-- Using cached msys-gcc-libs-11.2.0-3-x86_64.pkg.tar.zst.
-- Using cached msys-msys2-runtime-3.2.0-8-x86_64.pkg.tar.zst.
-- Using msys root at C:/Dev/vcpkg-downloads/tools/msys2/7e05e7aa09f1709f
-- Using cached msys-mingw-w64-i686-pkg-config-0.29.2-3-any.pkg.tar.zst.
-- Using cached msys-mingw-w64-i686-libwinpthread-git-9.0.0.6373.5be8fcd83-1-any.pkg.tar.zst.
-- Using msys root at C:/Dev/vcpkg-downloads/tools/msys2/9a1ec3f33446b195
-- Configuring x86-windows-dbg
-- Configuring x86-windows-rel
-- Building x86-windows-dbg
-- Installing x86-windows-dbg
-- Building x86-windows-rel
-- Installing x86-windows-rel
-- Installing: C:/Dev/vcpkg/packages/libiconv_x86-windows/share/libiconv/copyright
-- Performing post-build validation
-- Performing post-build validation done
Stored binary cache: C:\Users\bion\AppData\Local\vcpkg\archives\de\de68c15e08caefd1b503fb5b5407a8a2f69b8a5fec061c7ae6c1d59081e70a60.zip
Installing package libiconv[core]:x86-windows...
Elapsed time for package libiconv:x86-windows: 4.315 min

Total elapsed time: 4.362 min

The package libiconv provides CMake targets:

    find_package(iconv CONFIG REQUIRED)
    target_link_libraries(main PRIVATE Iconv::Charset)

PS C:\Dev\vcpkg> .\vcpkg.exe x-set-installed
The following packages will be removed:
    libiconv:x86-windows
Starting package 1/1: libiconv:x86-windows
Removing package libiconv:x86-windows...
Elapsed time for package libiconv:x86-windows: 23.49 ms
Restored 0 packages from C:\Users\bion\AppData\Local\vcpkg\archives in 120 us. Use --debug to see more details.

Total elapsed time: 27.63 ms

PS C:\Dev\vcpkg> .\scripts\tls12-download.exe github.com /microsoft/vcpkg-tool/releases/download/2022-03-03/vcpkg.exe vcpkg.exe
Downloading https://github.com/microsoft/vcpkg-tool/releases/download/2022-03-03/vcpkg.exe -> vcpkg.exe... done.
Validating signature... done.
PS C:\Dev\vcpkg> .\vcpkg.exe install libiconv --no-binarycaching
Computing installation plan...
The following packages will be built and installed:
    libiconv[core]:x86-windows -> 1.16#11
Detecting compiler hash for triplet x86-windows...
Starting package 1/1: libiconv:x86-windows
Building package libiconv[core]:x86-windows...
-- Using cached libiconv-1.16.tar.gz.
-- Cleaning sources at C:/Dev/vcpkg/buildtrees/libiconv/src/1.16-b0c41d3469.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source C:/Dev/vcpkg-downloads/libiconv-1.16.tar.gz
-- Applying patch 0002-Config-for-MSVC.patch
-- Applying patch 0003-Add-export.patch
-- Applying patch 0004-ModuleFileName.patch
-- Using source at C:/Dev/vcpkg/buildtrees/libiconv/src/1.16-b0c41d3469.clean
-- Getting CMake variables for x86-windows
-- Using cached msys-gzip-1.11-1-x86_64.pkg.tar.zst.
-- Using cached msys-bash-5.1.008-1-x86_64.pkg.tar.zst.
-- Using cached msys-autoconf-2.71-3-any.pkg.tar.zst.
-- Using cached msys-diffutils-3.8-2-x86_64.pkg.tar.zst.
-- Using cached msys-binutils-2.37-5-x86_64.pkg.tar.zst.
-- Using cached msys-libtool-2.4.6-9-x86_64.pkg.tar.xz.
-- Using cached msys-file-5.41-2-x86_64.pkg.tar.zst.
-- Using cached msys-zlib-1.2.11-1-x86_64.pkg.tar.xz.
-- Using cached msys-libbz2-1.0.8-3-x86_64.pkg.tar.zst.
-- Using cached msys-coreutils-8.32-2-x86_64.pkg.tar.zst.
-- Using cached msys-grep-3.0-2-x86_64.pkg.tar.xz.
-- Using cached msys-sed-4.8-2-x86_64.pkg.tar.zst.
-- Using cached msys-libpcre-8.45-1-x86_64.pkg.tar.zst.
-- Using cached msys-m4-1.4.19-2-x86_64.pkg.tar.zst.
-- Using cached msys-automake-wrapper-11-4-any.pkg.tar.zst.
-- Using cached msys-gawk-5.1.0-2-x86_64.pkg.tar.zst.
-- Using cached msys-mpfr-4.1.0-1-x86_64.pkg.tar.zst.
-- Using cached msys-gmp-6.2.1-1-x86_64.pkg.tar.zst.
-- Using cached msys-libreadline-8.1.001-1-x86_64.pkg.tar.zst.
-- Using cached msys-ncurses-6.2-2-x86_64.pkg.tar.zst.
-- Using cached msys-automake1.16-1.16.3-3-any.pkg.tar.zst.
-- Using cached msys-perl-5.32.1-2-x86_64.pkg.tar.zst.
-- Using cached msys-libcrypt-2.1-3-x86_64.pkg.tar.zst.
-- Using cached msys-pkg-config-0.29.2-4-x86_64.pkg.tar.zst.
-- Using cached msys-make-4.3-3-x86_64.pkg.tar.zst.
-- Using cached msys-findutils-4.8.0-1-x86_64.pkg.tar.zst.
-- Using cached msys-libintl-0.21-1-x86_64.pkg.tar.zst.
-- Using cached msys-libiconv-1.16-2-x86_64.pkg.tar.zst.
-- Using cached msys-gcc-libs-11.2.0-3-x86_64.pkg.tar.zst.
-- Using cached msys-msys2-runtime-3.2.0-8-x86_64.pkg.tar.zst.
-- Using msys root at C:/Dev/vcpkg-downloads/tools/msys2/7e05e7aa09f1709f
-- Using cached msys-mingw-w64-i686-pkg-config-0.29.2-3-any.pkg.tar.zst.
-- Using cached msys-mingw-w64-i686-libwinpthread-git-9.0.0.6373.5be8fcd83-1-any.pkg.tar.zst.
-- Using msys root at C:/Dev/vcpkg-downloads/tools/msys2/9a1ec3f33446b195
-- Configuring x86-windows-dbg
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:128 (message):
    Command failed: C:/Dev/vcpkg-downloads/tools/msys2/7e05e7aa09f1709f/usr/bin/bash.exe --noprofile --norc --debug -c "V=1 CPP='compile cl.exe -E' CC='compile cl.exe' CC_FOR_BUILD='compile cl.exe' CXX='compile cl.exe' RC='windres-rc rc.exe' WINDRES='windres-rc rc.exe' AR='ar-lib lib.exe' LD='link.exe -verbose' RANLIB=':' STRIP=':' NM='dumpbin.exe -symbols -headers' DLLTOOL='link.exe -verbose -dll' CCAS=':' AS=':' ./../src/1.16-b0c41d3469.clean/configure --build=x86_64-pc-mingw32 --host=i686-pc-mingw32 \"--enable-extra-encodings\" \"--without-libiconv-prefix\" \"--without-libintl-prefix\" \"--enable-relocatable\" \"ac_cv_prog_ac_ct_STRIP=:\" \"gl_cv_double_slash_root=yes\" \"ac_cv_func_memmove=yes\" \"--disable-silent-rules\" \"--verbose\" \"--enable-shared\" \"--disable-static\" \"--prefix=/C/Dev/vcpkg/installed/x86-windows/debug\" \"--bindir=\\${prefix}/../tools/libiconv/debug/bin\" \"--sbindir=\\${prefix}/../tools/libiconv/debug/sbin\" \"--libdir=\\${prefix}/lib\" \"--includedir=\\${prefix}/../include\" \"--datarootdir=\\${prefix}/share/libiconv\""
    Working Directory: C:/Dev/vcpkg/buildtrees/libiconv/x86-windows-dbg
    Error code: 1
    See logs for more information:
      C:\Dev\vcpkg\buildtrees\libiconv\config-x86-windows-dbg-out.log
      C:\Dev\vcpkg\buildtrees\libiconv\config-x86-windows-dbg-err.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_configure_make.cmake:815 (vcpkg_execute_required_process)
  ports/libiconv/portfile.cmake:29 (vcpkg_configure_make)
  scripts/ports.cmake:145 (include)


Error: Building package libiconv:x86-windows failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `git pull` and `.\vcpkg update`.
Then check for known issues at:
  https://github.com/microsoft/vcpkg/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+libiconv
You can submit a new issue at:
  https://github.com/microsoft/vcpkg/issues/new?template=report-package-build-failure.md&title=[libiconv]+Build+error
including:
  package: libiconv[core]:x86-windows -> 1.16#11
    vcpkg-tool version: 2022-03-03-6ef3648a2b7daf197fce82d936ff88ef98c26361
    vcpkg-scripts version: 5962cf5a3 2022-03-07 (2 hours ago)

Additionally, attach any relevant sections from the log files above.
PS C:\Dev\vcpkg>
```